presto-main/src/main/java/com/facebook/presto/execution/QueryManagerStats.java

```diff
@@ -28,6 +28,7 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Supplier;
 
+import static com.facebook.presto.execution.QueryState.QUEUED;
```

Comment:
I actually think `trackQueryStats()` methods in QueryManagerStats should NOT do  `queuedQueries.incrementAndGet();`. Having a query submitted doesn't mean it's already queued.  @tdcmeehan Do you know why they were done this way?

Reply:
> I actually think trackQueryStats() methods in QueryManagerStats should NOT do queuedQueries.incrementAndGet();. Having a query submitted doesn't mean it's already queued. 

I agree, that doesn't look right for both the methods. In the case of DispatchQuery, it is queued after the call to trackQueryStats, and in the case of SqlQueryManager, the query is not even in the queued state when trackQueryStats is called. It is called after waiting for resources finishes and query execution is about to start.